### ISSUE TITLE
several unused standard headers not reported as unused by `misc-include-cleaner`

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#71005]
```cpp
#include <cassert>
#include <cctype>
#include <cerrno>
#include <cfloat>
#include <ciso646>
#include <climits>
#include <clocale>
#include <cmath>
#include <csetjmp>
#include <csignal>
#include <cstdarg>
#include <cstdbool>
#include <cstddef>
#include <cstdint>
#include <cstdio>
#include <cstdlib>
#include <cstring>
#include <ctime>
#include <cuchar>
#include <cwchar>
#include <cwctype>

void f()
{
}
```

```
<source>:5:1: warning: included header ciso646 is not used directly [misc-include-cleaner]
    5 | #include <ciso646>
      | ^~~~~~~~~~~~~~~~~~
    6 | #include <climits>
<source>:13:1: warning: included header cstddef is not used directly [misc-include-cleaner]
   13 | #include <cstddef>
      | ^~~~~~~~~~~~~~~~~~
   14 | #include <cstdint>
<source>:14:1: warning: included header cstdint is not used directly [misc-include-cleaner]
   14 | #include <cstdint>
      | ^~~~~~~~~~~~~~~~~~
   15 | #include <cstdio>
<source>:19:1: warning: included header cuchar is not used directly [misc-include-cleaner]
   19 | #include <cuchar>
      | ^~~~~~~~~~~~~~~~~
   20 | #include <cwchar>
```
https://godbolt.org/z/ddGnr9Kc3

**Comment by firewave** [comment]
The latest code reports now reports most of them as unused but `<cassert>` and `<cstdbool>` are still not covered: https://godbolt.org/z/PW49vo14h
